### DISCUSSION TITLE
Error On Screen Rotation

### DISCUSSION THREAD

**Discussion by XichengSpencer** [oppia/oppia-android#5011] [Q&A (Other)] [answered]
What I did is change the Button to the Material Button in ResumeLessonFragment.

The error only happens when I rotate the screen and it cause the app to crash, if I stay in portrait/landscape it won't happen.

Here is the error message
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.oppia.android/org.oppia.android.app.resumelesson.ResumeLessonActivity}: 
java.lang.IllegalArgumentException: Wrong state class, expecting View State but received class com.google.android.material.button.MaterialButton$SavedState instead. 
This usually happens when two views of different type have the same id in the same hierarchy. This view's id is id/resume_lesson_start_over_button. Make sure other views do not use the same id.

![Screenshot 2023-06-02 124639](https://github.com/oppia/oppia-android/assets/74568012/12368828-2b5b-4796-8acc-68cdaf3529cd)

I check the usage of the button. There is only a single id reference. It is not used in code, but it is used in several layout files. Since I only make a change button to the material button for 1 layout file, could it be the inconsistent type across the different layout files caused this problem? If I change them all will it be fixed?

**Comment by adhiamboperes** [comment, marked as answer]
Hi @XichengSpencer. the error message is assentially correct. You have changed the button class in the default layout but not in the landscape or sw-600dp layouts. See below:

<img width="1425" alt="Screenshot 2023-06-05 at 08 17 50" src="https://github.com/oppia/oppia-android/assets/59600948/3ce964ea-1950-4e59-8bb5-f05c3e1576e9">
